### PR TITLE
Overhaul

### PR DESCRIPTION
**Pull request by MSWS**

### CODE REVIEW

File: EGO Forum Enhancement.user.js [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+ * @param {number} post_title Steam ID to lookup
+ */
+function addLookupButton(div, post_title) {
+    var steam_id_unknown = post_title.match(/^.* - .* - (?<game_id>[\w\d\/\[\]\-\.:]*)$/);
+    if (steam_id_unknown)
+        createButton("https://steamid.io/lookup/" + steam_id_unknown.groups.game_id, "Lookup ID", div);
+}
+
+/**
+ * Adds a "Move to Complted" button to the div
```
Comment (blankdvth) [inline EGO Forum Enhancement.user.js:67]: `Completed` is misspelled. This comment may be redundant if the next one is implemented.

File: EGO Forum Enhancement.user.js [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+    text.innerHTML = str;
+    text.style.color = "rgba(255,0,0,0.25)";
+    text.style.fontSize = "100px";
+    text.style.position = "fixed";
+    text.style.top = top;
+    text.style.left = "50%";
+    text.style.transform = "translateX(-50%)"
+    text.style.pointerEvents = "none";
+    text.style.zIndex = "999";
+}
```
Comment (blankdvth) [inline EGO Forum Enhancement.user.js:276]: Would be ideal to move this up to near the top of the file where all the other helper functions are. The file follows a nice layout right now, although this particular function is an outlier.

File: EGO MAUL Enhancement.user.js [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+    var button = document.createElement("button");
+    button.classList.add("btn", "btn-default");
+    button.innerHTML = text;
+    button.onclick = callback;
+    button.style.marginRight = "4px";
+    return button
+}
+
+/**
+ * Creates a preset button {@see createPresetButton}
```
Comment (blankdvth) [inline EGO MAUL Enhancement.user.js:64]:
```suggestion
 * Creates a link button
```

This function creates a button that leads to a link (href), as opposed to a preset. The description is misleading here.

File: EGO MAUL Enhancement.user.js [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+    for (let expandingElement of expandingElementList) {
+        let element = expandingElement.childNodes[1];
+        let adminsOrReason = element.childNodes[14].textContent;
+        let adminsElement;
+        if (adminsOrReason === "Admins Online:") {
+            // Admins Online field, fetch 17th
+            adminsElement = element.childNodes[17];
+        } else {
+            // Reason field is given, fetch 24th
+            adminsElement = element.childNodes[24];
```
Comment (blankdvth) [inline EGO MAUL Enhancement.user.js:208]: Could we modify this to use querySelector as opposed to indexes of childNodes? Using indexes of childNodes can easily result in issues down the line if MAUL updates the format, or even if another script inserts something into the element.

File: EGO MAUL Enhancement.user.js [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+        }
+        for (let admin of adminsElement.textContent.split(", ")) {
+            let id = knownAdmins[admin];
+            if (id == undefined)
+                continue;
+            adminsElement.innerHTML = adminsElement.innerHTML.replaceAll(admin, `<a href="https://maul.edgegamers.com/index.php?page=home&id=${knownAdmins[admin]}">${admin}</a>`);
```
Comment (blankdvth) [inline EGO MAUL Enhancement.user.js:214]: `replaceAll` shouldn't be needed here as an admin name should only occur once within a given Admins Online list.

**Review by blankdvth — CHANGES_REQUESTED** [review]
Really nice changes! Love the refactor and docstrings for the larger scripts. There's just a few issues/suggestions I picked up on while reading through it.

P.S. GPL license does seem overkill for such a project, although I don't mind either way.

File: EGO Forum Enhancement.user.js [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+/**
+ * Adds a "Move to Complted" button to the div
+ * @param {HTMLDivElement} div Div to add to
+ * @param {string} url URL to move to
+ */
+function addMoveButton(div, url) {
+    var post_id = url.match(/threads\/(?<post_id>\d+)/)
+    if (post_id)
+        createButton("https://www.edgegamers.com/threads/" + post_id.groups.post_id + "/move?move_to_completed", "Move to Completed", div, "_self");
+}
```
[the remark this answers is not in the document]
Comment (blankdvth) [reply EGO Forum Enhancement.user.js:75]: [Duplicate](https://github.com/blankdvth/eGOScripts/pull/2#discussion_r1067180552)

File: EGO MAUL Enhancement.user.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (MSWS) [reply EGO MAUL Enhancement.user.js]: After switching to .replace this should be fixed, as replace only replaces the first instance.

File: EGO Forum Enhancement.user.js [hunk 7]
```diff
@@ -0,0 +1,301 @@
...
+    }
+    if (isLeadership(breadcrumbs)) // LE Forums
+        handleLeadership();
+}
+
+/**
+ * Adds "View Bans" or "Lookup ID" button on report/contest threads.
+ * TODO: Add support for other game IDs
+ */
+function handleBanReport() {
```
[the remark this answers is not in the document]
Comment (blankdvth) [reply EGO Forum Enhancement.user.js:247]: Unneeded

File: EGO Forum Enhancement.user.js [hunk 8]
```diff
@@ -0,0 +1,301 @@
...
+    }
+}
+
+/**
+ * Checks if a given breadcrumbs string contains LE threads
+ * @param {string} str 
+ * @returns true if LE, false otherwise
+ */
+function isLeadership(str) {
+    return str.match(/(Leadership|Report a Player|Report Completed)/);
```
[the remark this answers is not in the document]
Comment (blankdvth) [reply EGO Forum Enhancement.user.js:217]: Moving to an issue (#3) as this is more of a matter of opinion/preference.

File: EGO Forum Enhancement.user.js [hunk 9]
```diff
@@ -0,0 +1,301 @@
...
+
+    dropdown.setAttribute('data-xf-key', '3');
+    dropdown.setAttribute('data-xf-click', 'menu');
+    dropdown.setAttribute('data-menu-pos-ref', '< .p-navEl');
+    dropdown.setAttribute('class', 'p-navEl-splitTrigger');
+    dropdown.setAttribute('role', 'button');
+    dropdown.setAttribute('tabindex', '0');
+    dropdown.setAttribute('aria-label', 'Toggle expanded');
+    dropdown.setAttribute('aria-expanded', 'false');
+    dropdown.setAttribute('aria-haspopup', 'true');
```
[the remark this answers is not in the document]
Comment (blankdvth) [reply EGO Forum Enhancement.user.js:119]: Moved to issue (#4), resolving as something that will be worked on separate from this P.R.

**Review by blankdvth — APPROVED** [review]
Looks good, approving changes. Remaining conversations moved to issues that will be worked on separately.

Admins Online overlap seems to still be an issue, but is more of an edge case and can be looked into later. (#5)

**Pull request merged**